### PR TITLE
Rip useless names out of RawDef

### PR DESCRIPTION
**Pull request by kazimuth**
# Description of Changes

Addresses https://github.com/clockworklabs/SpacetimeDB/issues/1893
Does NOT address https://github.com/clockworklabs/SpacetimeDB/issues/1891 since the special names turned out to be used throughout the codebase and I think addressing that will require a separate big PR.

While we're reworking naming, we may want to revisit the naming scheme here. It has a problem.
Consider the following declaration:

```rust
#[spacetimedb(table)]
#[derive(Clone, Debug)]
#[spacetimedb(index(name = "idx1", btree(columns = [a, b]))]
#[spacetimedb(index(name = "idx2", btree(columns = [a_b]))]
pub struct EdgeCase {
    pub a: u32,
    pub b: u32,
    pub a_b: u64,
}
```

These indexes will BOTH receive the generated name `"idx_EdgeCase_btree_a_b"`. Weird errors result.

(`idx1` and `idx2` are now referred to as `accessor_name`s and are stored separately from the name of the index itself.)

It would be better if we used out-of-band characters in the index name, like, say, `"EdgeCase.index.btree([a,b])"`. However, if we do this, we will need to relax the `Identifier` validation rules for indexes, since normally those will reject special characters. Also, it means that these identifiers can no longer be used directly as identifiers in SQL. @mamcx, I would welcome your opinion on this. I think it's fine, since system table queries can just wrap names in quotes.

We could also use a more elaborate mangling scheme, but this may penalize readability of the system tables.

# API and ABI breaking changes
ABI breaking.

# Expected complexity level and risk

Reduces complexity overall so let's say 2->1.

# Testing

Fixed all relevant tests.

### CODE REVIEW

**Review by RReverser — APPROVED** [review]
Thanks, LGTM. My only comments would be what's already covered by your TODOs.

**Comment by mamcx** [comment]
I don't understand why it will generate `"idx_EdgeCase_btree_a_b"`. I expect it to be `"idx1_EdgeCase_btree_a_b"/"idx2_EdgeCase_btree_a_b"`

**Review by coolreader18 — APPROVED** [review]

**Comment by kazimuth** [comment]
See also https://github.com/clockworklabs/SpacetimeDBPrivate/pull/915

File: crates/bindings-csharp/Codegen.Tests/fixtures/server/snapshots/Module#FFI.verified.cs [hunk 1]
```diff
@@ -231,7 +229,7 @@ internal BTreeViewsUniqueIndex(BTreeViews handle)
             internal BTreeViewsUniqueIndex Id => new(this);
 
             internal sealed class LocationIndex()
-                : SpacetimeDB.Internal.IndexBase<global::BTreeViews>("bt_BTreeViews_Location")
+                : SpacetimeDB.Internal.IndexBase<global::BTreeViews>("index(,[X,Y])")
```
Comment (kazimuth) [inline crates/bindings-csharp/Codegen.Tests/fixtures/server/snapshots/Module#FFI.verified.cs:232]: ah, this is wrong, wait.

**Comment by kazimuth** [comment]
I think the remaining test failure here could be related to https://github.com/clockworklabs/SpacetimeDB/pull/1987 so I'm going to work on that first.

File: crates/core/src/db/datastore/locking_tx_datastore/datastore.rs [hunk 2]
```diff
@@ -1449,17 +1449,17 @@ mod tests {
...
+            ConstraintRow { constraint_id: 6, table_id: ST_CONSTRAINT_ID.into(), unique_columns: col(0), constraint_name: "constraint.unique(st_constraint,[constraint_id])", },
+            ConstraintRow { constraint_id: 7, table_id: ST_CLIENT_ID.into(), unique_columns: col_list![0, 1], constraint_name: "constraint.unique(st_client,[identity,address])", },
+            ConstraintRow { constraint_id: 8, table_id: ST_VAR_ID.into(), unique_columns: col(0), constraint_name: "constraint.unique(st_var,[name])", },
+            ConstraintRow { constraint_id: 9, table_id: ST_SCHEDULED_ID.into(), unique_columns: col(0), constraint_name: "constraint.unique(st_scheduled,[schedule_id])", },
+            ConstraintRow { constraint_id: 10, table_id: ST_SCHEDULED_ID.into(), unique_columns: col(1), constraint_name: "constraint.unique(st_scheduled,[table_id])", },
+            ConstraintRow { constraint_id: 11, table_id: ST_ROW_LEVEL_SECURITY_ID.into(), unique_columns: col(1), constraint_name: "constraint.unique(st_row_level_security,[sql])", },
```
Comment (kazimuth) [inline crates/core/src/db/datastore/locking_tx_datastore/datastore.rs:1462]: Here are examples of the new auto-generated names in action.

Comment (kazimuth) [inline crates/core/src/db/datastore/locking_tx_datastore/datastore.rs:1462]: Here are examples of the new auto-generated default names in action.

Comment (kazimuth) [inline crates/core/src/db/datastore/locking_tx_datastore/datastore.rs:1462]: Here are examples of the new auto-generated default names in action.

Comment (cloutiertyler) [reply crates/core/src/db/datastore/locking_tx_datastore/datastore.rs:1462]: Please use the Postgres standard naming scheme: https://stackoverflow.com/a/4108266/1445441

**Review by cloutiertyler — CHANGES_REQUESTED** [review]
Please see comment.

File: crates/core/src/db/datastore/locking_tx_datastore/datastore.rs (hunk 2 above)
Comment (kazimuth) [reply crates/core/src/db/datastore/locking_tx_datastore/datastore.rs:1462]: That is sensitive to name collisions but it's easy to adopt.

**Review by cloutiertyler — APPROVED** [review]
Test suite is failing, but this now LGTM from my perspective on names

**Pull request merged**